### DISCUSSION TITLE
MQTTX web missing features from Desktop App

### DISCUSSION THREAD

**Discussion by unai001** [emqx/MQTTX#1677] [Q&A] [answered]
### The webapp has some key differences from the desktop application

- Desktop app: Has **mqtt** and **mqtts**  **ws** and **wss** hosts

![lgCfs30W0u](https://github.com/emqx/MQTTX/assets/79275029/cc33013d-a4bc-469c-926e-fa637e9b2009)

- While the web app only has **ws** and **wss**

![image](https://github.com/emqx/MQTTX/assets/79275029/371b9b3a-98b0-4960-acbc-1328ad4a78bb)

- Also, to import and export configurations, the desktop hass has the data backup option, that generates a JSON file

![image](https://github.com/emqx/MQTTX/assets/79275029/c7b5ce96-e118-4945-989a-734c551a5ed8)

- But the webapp is missing a lot of settings, including the advanced ones that i need

![image](https://github.com/emqx/MQTTX/assets/79275029/913b9634-8a2b-422c-a5bc-6b9a08a7c252)

Im self hosting the web application via docker, followed https://mqttx.app/docs/web/get-started

**Comment by Red-Asuka** [comment, marked as answer]
Hello, in fact, we have already given a [detailed explanation](https://mqttx.app/docs/web/get-started#differences-between-web-and-desktop) about this in our documentation. The core difference lies in the fact that the operating environments for web and desktop are not identical, hence they utilize distinct codes leading to some variations. The core functionalities offered by web align with that of desktop, albeit it lags slightly in the provision of new features. We'll be persistently updating to maintain the synchronization of the latest features. For complex and heavy-duty testing scenarios, we currently recommend the use of the desktop version.

**Reply by Red-Asuka** [reply]
We're planning on introducing a data import/export feature in the next release of MQTTX Web. In fact, this feature is already available on our [online website](http://mqtt-client.emqx.com/). However, we've not yet released the version for self-deployment via Docker images, which will be available shortly. Please note during use that the data from the Web version is not interchangeable with the Desktop version.

**Reply by unai001** [reply]
Thanks for the fast reply mate, the application im developing can work with webSocket, but would be nice to have mqtt and mqtts

Also, until the import/export of JSON settings, i saw the settings themselves get stored in local storage. Do you think it would be a good idea to add a login to the mqtt broker so it automatically puts one settings or others depending on the user?

**Reply by Red-Asuka** [reply]
Due to limitations brought by browser environments, MQTT and MQTTS are currently not supported. If you need to use MQTT and MQTTS in scenarios, I'd recommend testing it via MQTTX Desktop or MQTTX CLI.

As for MQTTX Web, all the data will be stored on your local computer, with no cloud operations or data processing conducted by us. Any message data is only related to the service you are connected to. We haven't planned any feature concerning log-in and data cloud backup for now, as it may involve users' private data. But rest assured, we are closely monitoring users' feedback and will decide whether to introduce such features based on their inputs.

**Reply by ysfscream** [reply]
Hi, the login system has been scheduled for future versions, and we will support cloud synchronization for users.

In version 2.0, we have revamped the codebase, allowing some functionality to be shared between the web and desktop versions. Therefore, in the future, the web version may have functionalities consistent with the desktop version.

Thank you for your feedback.
